language/move-lang/src/naming/uses.rs

```diff
@@ -50,63 +61,109 @@ impl<'a> Context<'a> {
         }
     }
 
-    fn add_usage(&mut self, uses: &ModuleIdent, loc: Loc) {
-        if self.current_module.as_ref().unwrap() == uses || !self.modules.contains_key(uses) {
+    fn add_neighbor(&mut self, mident: ModuleIdent, dep_type: DepType, loc: Loc) {
+        let current_mident = self.current_module.clone().unwrap();
+        if current_mident == mident || !self.modules.contains_key(&mident) {
             return;
         }
+        let (upstream, downstream) = match dep_type {
```

Comment:
I don't understand this upstream/downstream terminology as it isn't really a graph related term

Perhaps `(node, new_neighbor)`?